rfc/0052-sdk3-full-text-search.md

```diff
@@ -566,7 +566,275 @@ A query that matches nothing.
+
+The current size limit for the vector is 2048 elements, but this will not be enforced on the SDK side.
+The vector must be non-empty though, and if not the SDK will raise `InvalidArgument`. 
+
+It will also support the following parameters, exposed in the same way as the SDK exposes traditional FTS query parameters.
+In some SDKs this is as fluent-style methods on `VectorQuery` itself, but a `VectorQueryOptions` block as an optional parameter on `Vector.create()` would be more SDK3-idiomatic.
+The SDK should follow its existing convention for FTS parameters.
+
+* `uint32 numCandidates`.  Sent as a `k` number field in the JSON.  We name the field differently in the SDK for the same reason as `knn_operator`.  If not set by the user, FTS PM requests that the SDK send a default value of 3.  It controls how many results are returned and must be >= 1.  If < 1 the SDK will raise `InvalidArgument`.
+* `float boost`.  Sent as a `boost` number field in the JSON.  If not set, the field is not sent.
+
+`VectorQuery` does not extend the `SearchQuery` interface, which is now reserved for traditional FTS queries.
+
+### SearchRequest
+`SearchRequest` creation is platform-idiomatic.  There need to be two contruction options, allowing it to be created from either a `VectorSearch` or a `SearchQuery`.
+
+```
+SearchRequest.vectorSearch(VectorSearch vectorSearch, [SearchOptions searchOptions])
+SearchRequest.searchQuery(SearchQuery searchQuery, [SearchOptions searchOptions])
+```
+
+It will also support the following fluent-style methods, allowing one (and only one) `SearchQuery` or `VectorSearch` to be added:
+
+* `searchQuery(SearchQuery searchQuery)`.  Note that this intentionally does not take a SearchOptions - that can only be specified during construction.  If the user has already specified a `SearchQuery`, either at construction time or via another call to `SearchRequest.searchQuery()`, the SDK needs to raise an `InvalidArgumentException`.
+* `vectorSearch(VectorSearch vectorSearch)`.  Note that this intentionally does not take a SearchOptions - that can only be specified during construction.  If the user has already specified a `VectorSearch`, either at construction time or via another call to `SearchRequest.vectorSearch()`, the SDK needs to raise an `InvalidArgumentException`.
+
+These fluent-style methods are proposed because, at least in some SDKs, FTS follows a fluent style.
+If this is completely unidiomatic to an SDK, this alternative single constructor can be considered:
+
+```
+SearchRequest.searchRequest([SearchQuery searchQuery], [VectorSearch vectorSearch], [SearchOptions searchOptions])
+```
+
+In this case, at least one of `SearchQuery` and `VectorSearch` must be provided and the SDK must raise `InvalidArgumentException` if not.
+
+`SearchRequest` does not extend the `SearchQuery` interface, which is now reserved for traditional FTS queries.
+
+### showrequest
```

Comment:
Is this behaviour only part of the new `search()` methods, or should it be applied to the existing `searchQuery` method?  It says "for all queries", but I think it's only for the new method, which should be clarified.

Reply:
That's correct, it applies to all queries from the new method.  I'll clarify it.